part1/anecdote/src/App.jsx

```diff
@@ -0,0 +1,59 @@
+import { useState } from 'react'
+
+const Button = ({onClick, text}) => <button onClick={onClick}>{text}</button>
+
+const Display = ({anecdotes, points}) => {
+  return (
+    <div>
+      <p>{anecdotes}</p>
+      <p>has {points} points</p>
+    </div>
+  )
+}
+
+const App = () => {
+
+  function getRandomInt(max) {
+    return Math.floor(Math.random() * max);
+  }
+
+  const handleNextAnecdote = () => {
+    return setSelected(getRandomInt(8))
+  }
+
+  const handleVote = () => {
+    const copy = [...points]
+    copy[selected] += 1
+    setPoints(copy)
+  }
+
+  const anecdotes = [
+    'If it hurts, do it more often.',
+    'Adding manpower to a late software project makes it later!',
+    'The first 90 percent of the code accounts for the first 90 percent of the development time...The remaining 10 percent of the code accounts for the other 90 percent of the development time.',
+    'Any fool can write code that a computer can understand. Good programmers write code that humans can understand.',
+    'Premature optimization is the root of all evil.',
+    'Debugging is twice as hard as writing the code in the first place. Therefore, if you write the code as cleverly as possible, you are, by definition, not smart enough to debug it.',
+    'Programming without an extremely heavy use of console.log is same as if a doctor would refuse to use x-rays or blood tests when diagnosing patients.',
+    'The only way to go fast, is to go well.'
+  ]
+   
+  const [selected, setSelected] = useState(0)
+  const [points, setPoints] = useState(new Uint8Array(8))
```

Comment:
an 8 bit integer at max can be 255. in JS just pass in `[]`
```suggestion
  const [points, setPoints] = useState([])
```

JS uses sparse arrays which is kinda spicy, if you access any element, even if it doesn't exist itll grow to be that size lol https://www.freecodecamp.org/news/sparse-and-dense-arrays-in-javascript/